Add week-based calculations, start of week support, DAYOFWEEK, DAYNAME functions, and WEEK-based DATEDIFF functionality (#292)
bodo-ai/PyDough: documentation/usage.md

  key.DESC(),
)
```

The following configs are used in the behavior of `DAYOFWEEK` and`DATEDIFF`:

5. `start_of_week` (default=`DayOfWeek.SUNDAY`): Determines which day is considered the first day of the week. This affects functions like `DATEDIFF` and `DAYOFWEEK`. The value must be one of the following `DayOfWeek` enum values:

   - `DayOfWeek.SUNDAY` (default)
   - `DayOfWeek.MONDAY`  
   - `DayOfWeek.TUESDAY`
   - `DayOfWeek.WEDNESDAY`
   - `DayOfWeek.THURSDAY`
   - `DayOfWeek.FRIDAY`
   - `DayOfWeek.SATURDAY`

   The `DayOfWeek` enum is defined in the `pydough.configs` module.
6. `start_week_as_zero` (default=True): if True, then the first day of the week is considered to be 0. If False, then the first day of the week is considered to be 1.

```py
import pydough
from pydough.configs import DayOfWeek

# The configs of the active session
configs = pydough.active_session.config

# Set the start of the week to Monday

> configs.start_of_week = DayOfWeek.MONDAY

Author: I've done it on the previous lines. I did not do it for DATEDIFF and DATETIME as`start_week_as_zero` config does not these functions.

> configs.start_of_week = DayOfWeek.MONDAY

Reviewer: Ahh, sorry didn't see

  key.DESC(),
)
```

The following configs are used in the behavior of `DAYOFWEEK`, `DATETIME`, and `DATEDIFF`:

> 5. `start_of_week` (default=`DayOfWeek.SUNDAY`): Determines which day is considered the first day of the week. This affects functions like 

Reviewer 2: ```suggestion
5. `start_of_week` (default=`DayOfWeek.SUNDAY`): Determines which day is considered the first day of the week. This affects the following functions:
```

> - `DAYOFWEEK` : An integer that returns the number of days since the start of the week. Start of week is relative to the `start_of_week` config.

Reviewer 2: ```suggestion
- `DAYOFWEEK` : A function that returns the number of days since the start of the week. Start of week is relative to the `start_of_week` config.
```
